File: tests-playwright/utils/page-manager.ts
import { Page } from '@playwright/test';
import { LoginPage } from '../pages/login-page';
import { InventoryPage } from '../pages/inventory-page';
import { CartPage } from '../pages/cart-page';
import { CheckoutPage } from '../pages/checkout-page';
import { CheckoutOverviewPage } from '../pages/checkout-overview-page';
import { CheckoutCompletePage } from '../pages/checkout-complete-page';

export class PageManager {
  public loginPage: LoginPage;
  public inventoryPage: InventoryPage;
  public cartPage: CartPage;
  public checkoutPage: CheckoutPage;
  public checkoutOverviewPage: CheckoutOverviewPage;
  public checkoutCompletePage: CheckoutCompletePage;

  constructor(page: Page) {
    this.loginPage = new LoginPage(page);
    this.inventoryPage = new InventoryPage(page);
    this.cartPage = new CartPage(page);
    this.checkoutPage = new CheckoutPage(page);
    this.checkoutOverviewPage = new CheckoutOverviewPage(page);
    this.checkoutCompletePage = new CheckoutCompletePage(page);
  }
}

File: tests-csharp/Utils/PageManager.cs
using Microsoft.Playwright;
using Tests.CSharp.Pages;

namespace Tests.CSharp.Utils;

public class PageManager
{
    private readonly IPage _page;

    public LoginPage LoginPage { get; }
    public InventoryPage InventoryPage { get; }
    public CartPage CartPage { get; }
    public CheckoutPage CheckoutPage { get; }
    public CheckoutOverviewPage CheckoutOverviewPage { get; }
    public CheckoutCompletePage CheckoutCompletePage { get; }

    public PageManager(IPage page)
    {
        _page = page;
        LoginPage = new LoginPage(page);
        InventoryPage = new InventoryPage(page);
        CartPage = new CartPage(page);
        CheckoutPage = new CheckoutPage(page);
        CheckoutOverviewPage = new CheckoutOverviewPage(page);
        CheckoutCompletePage = new CheckoutCompletePage(page);
    }
}

